### PR TITLE
UT_Alignment_12d_4

### PR DESCRIPTION
**Pull request by LeeGregory12d**
# IFC4.3 RC2 Unit test

New UT Alignment with left arc and leading and trailing Clothoid transitions

## Review requested
 Clothoid transition
 
@pjanck

## Instructions
 Added new UT UT_Alignment_12d_4 with supporting documentation

### CODE REVIEW

File: IFC 4.3/UT_Alignment_12d_4/UT_Alignment_12d_4.ifc [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+#22 = IFCRELAGGREGATES('3dGRQ8is19OxGu3HAN$2Kr', #2, 'SiteContainer', 'SiteContainer For All Models', #19, (#21));
+#23 = IFCRELCONTAINEDINSPATIALSTRUCTURE('3SjitDIVb18xVX8J0OQhuE', #2, 'Building Container', 'BuildingContainer for Elements', (#24), #21);
+#24 = IFCALIGNMENT('1lDwpGZZ196A9C_7$_fNcZ', $, 'RS01 with clothoid no VG', '12d Alignment String', 'natural clothoid', $, $, $, $);
+#25 = IFCRELAGGREGATES('3E9jkWubnEhfuo6SbSqGVt', #2, '', '', #24, (#26));
+#26 = IFCALIGNMENTHORIZONTAL('0Rwb_gu3v6Zv174iuvrRAg', #2, 'Horizontal Alignment', '12d Horizontal Alignment', $, $, $, 0., (#28, #30, #32, #34, #36));
+#27 = IFCRELNESTS('2WxrrInhr9f9ee7WcPux5u', #2, '', '', #26, (#28, #30, #32, #34, #36));
+#28 = IFCALIGNMENTHORIZONTALSEGMENT($, $, #29, IFCPLANEANGLEMEASURE(0.), 0., 0., IFCPOSITIVELENGTHMEASURE(38.2922833180901), $, .LINE.);
+#29 = IFCCARTESIANPOINT((0., 0.));
+#30 = IFCALIGNMENTHORIZONTALSEGMENT($, $, #31, IFCPLANEANGLEMEASURE(0.), -0., -150., IFCPOSITIVELENGTHMEASURE(20.), $, .CLOTHOID.);
+#31 = IFCCARTESIANPOINT((38.2922833180901, 0.));
```
Comment (TLiebich) [inline IFC 4.3/UT_Alignment_12d_4/UT_Alignment_12d_4.ifc:38]: start / end radius - shouldn't it be 0., 150  (and not -0., -150) ?

File: IFC 4.3/UT_Alignment_12d_4/UT_Alignment_12d_4.ifc [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+#32 = IFCALIGNMENTHORIZONTALSEGMENT($, $, #33, IFCPLANEANGLEMEASURE(6.66666666671789E-2), 150., 150., IFCPOSITIVELENGTHMEASURE(74.1184532245796), $, .CIRCULARARC.);
+#33 = IFCCARTESIANPOINT((58.2833962580021, 4.44303370933568E-1));
+#34 = IFCALIGNMENTHORIZONTALSEGMENT($, $, #35, IFCPLANEANGLEMEASURE(5.60789688156486E-1), -150., -0., IFCPOSITIVELENGTHMEASURE(30.), $, .CLOTHOID.);
+#35 = IFCCARTESIANPOINT((152.343929366576, 40.6906546571354));
```
Comment (TLiebich) [inline IFC 4.3/UT_Alignment_12d_4/UT_Alignment_12d_4.ifc:42]: same here -> 150., 0.

Comment (TLiebich) [reply IFC 4.3/UT_Alignment_12d_4/UT_Alignment_12d_4.ifc:42]: #35 = IFCCARTESIANPOINT((152.343929366576, 40.6906546571354)); 
looks wrong - is identical as in line#37

would assume the values as shown in the PNG -> 128.069...  23.085...

**Review by TLiebich — COMMENTED** [review]
minor issues, particular the clothoid looks wrong

**Comment by larswik** [comment]
Right now it seems as if the uploaded file contains things like "<<<<<<< Updated upstream" etc. Can you please upload the alignment file?

**Comment by LeeGregory12d** [comment]
Sorry - I will re-upload the unadulterated file

**Review by larswik — APPROVED** [review]

**Comment by LeeGregory12d** [comment]
UT moved to repository IFC-infra-unit-test

**Pull request closed without merging**